src/platform/backends/qemu/qemu_vm_process_spec.cpp

```diff
@@ -17,13 +17,49 @@
 
 #include "qemu_vm_process_spec.h"
 
+#include <multipass/format.h>
+#include <multipass/logging/log.h>
 #include <shared/linux/backend_utils.h>
 
 namespace mp = multipass;
+namespace mpl = multipass::logging;
 
-mp::QemuVMProcessSpec::QemuVMProcessSpec(const mp::VirtualMachineDescription& desc, const QString& tap_device_name,
-                                         const QString& mac_addr)
-    : desc(desc), tap_device_name(tap_device_name), mac_addr(mac_addr)
+namespace
+{
+constexpr auto default_machine_type = "pc-i440fx-xenial";
+} // namespace
+
+/*
+ * Qemu will often fail to resume a VM that was run with a different command line.
+ * To support backward compatibility, we version each Qemu command line iteration.
+ *
+ * Versions:
+ *  2 - change how disk image is specified, enabling image discard feature
+ *      Replaced: "-hda image.img"
+ *      With:     "-device virtio-scsi-pci,id=scsi0
+ *                 -drive file=image.img,if=none,format=qcow2,discard=unmap,id=hda
+ *                 -device scsi-hd,drive=hda,bus=scsi0.0"
+ *  1 - changed how cloud-init ISO was specified:
+ *      Replaced: "-drive file=cloud-init.iso,if=virtio,format=raw,snapshot=off,read-only"
+ *      With:     "-cdrom cloud-init.iso"
+ *      Note this was originally encompassed in the metadata as "use_cdrom" being true.
+ *  0 - original
+ */
```

Comment:
I would prefer we just save the whole `qemu` command line used on startup and discard on shutdown, this way we'd achieve 100% the same with much less complexity…

Reply:
I thought the tap network device interface name could change between suspend/resume. But aside from that, sure, that can be done.